### ISSUE TITLE
bazel_dep allows missing version with poor error

### ISSUE THREAD

**Issue by keith** [bazelbuild/bazel#17126]
### Description of the bug:

If you specify this in your MODULE.bazel:

```
bazel_dep(
    name = "apple_support",
)
```

bazel errors with:

```
ERROR: Error computing the main repository mapping: module not found in registries: apple_support@_
```

I imagine omitting a version is supported for overrides(?), but i think ideally the error would hint to the user that they are missing the version.

### What is the output of `bazel info release`?

release 6.0.0rc4

**Comment by iancha1992** [comment]
@bazel-io fork 6.4.0

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 6.4.0 RC1](https://releases.bazel.build/6.4.0/rc1/index.html). Please test out the release candidate and report any issues as soon as possible. Thanks!
